internals/cli/masker/writer.go

```diff
@@ -0,0 +1,204 @@
+package masker
+
+import (
+	"io"
+	"sync"
+	"time"
+)
+
+type Matcher interface {
```

Comment:
Please add some docs to this exported interface.

Reply:
Could you document the functions as well?

Reply:
![document-all-things](https://user-images.githubusercontent.com/9931648/57929870-bb19fe00-78b4-11e9-8aff-2e1e526c4c25.jpg)

Reply:
Or shall I just privatize the whole interface? There is no reason to export it anyway.

Reply:
Currently, you can create your own implementation of `Matcher` and construct an instance of `MaskedWriter` which uses this own implementation.

We could make it private. If we do, `MaskedWriter.Matchers` should be made private as well.
It is then no longer possible to create a valid `MaskedWriter` using the struct outside the package. Do we want to make the other fields private as well then?

I'd prefer to keep it as is and document the functions.

Reply:
You'll still be able to do that if it's private. As long as your implementation satisfies the interface, you can make it work. If it's export it means you can use this interface outside of the package. Meaning that it would be useful for someone to use this interface. I don't think it is, therefore it would be better to make it private.

Reply:
godoc is really clear now :+1: